eessi/testsuite/hooks.py

```diff
@@ -282,16 +282,58 @@ def _assign_one_task_per_gpu(test: rfm.RegressionTest):
     log(f'num_cpus_per_task set to {test.num_cpus_per_task}')
     log(f'num_tasks set to {test.num_tasks}')
 
+def _set_or_append_valid_systems(test: rfm.RegressionTest, valid_systems: str):
+    """
+    Sets test.valid_systems based on the valid_systems argument.
+    - If valid_systems is an empty string, test.valid_systems is set equal to eessi.testsuite.constants.INVALID_SYSTEM
+    - If no test.valid_system was set yet, or it was at the default value ['*'], it is overwritten by [valid_system]
+    - If a test.valid_system was already set, valid_system is appended to it, which allows adding requests for multiple
+    partition features by different hooks.
+    """
+
+    # This indicates an invalid test that always has to be filtered
+    if valid_systems == '':
+        test.valid_systems = [INVALID_SYSTEM]
+        return
+
+    # test.valid_systems wasn't set yet, so set it
+    if len(test.valid_systems) == 0:
+        # test.valid_systems is empty, meaning all tests are filtered out. This hook shouldn't change that
+        pass
+    # test.valid_systems still at default value, so overwrite
+    elif len(test.valid_systems) == 1 and test.valid_systems[0] == '*':
+        test.valid_systems = [valid_systems]
+    # test.valid_systems was set before, so append
+    elif len(test.valid_systems) == 1:
+        test.valid_systems[0] = f'{test.valid_systems[0]} {valid_systems}'
+    else:
```

Comment:
@smoors or should we just do a `pass` in this case as well? If someone puts `valid_systems=[A,B]` on command line, we should probably just let that be?

Reply:
maybe print a warning in that case? something like
```python
import warnings
warning.warn('valid_systems has multiple items, which is not supported by this hook. make sure to handle filtering yourself')
```

Reply:
Done, and adapted the function description accordingly